src/RedirectMiddleware.php

```diff
@@ -77,7 +77,7 @@ public function checkRedirect(
         array $options,
         ResponseInterface $response
     ) {
-        if (\substr($response->getStatusCode(), 0, 1) != '3'
+        if (\strpos((string) $response->getStatusCode(), '3') !== 0
```

Comment:
why is this change needed? 

substr vs strpos

Reply:
Micro optimization (`strpos` is roughly 20% faster and consuming less memory) in combination with fixing a PHPStan error.

Feel free to let me know if you want me to revert this change.